include/NovelRTInterop/Maths/GeoVector2F_t.h

```diff
@@ -9,224 +13,162 @@
-  return normal;
-}
-
-inline float GeoVector2F_getLength(GeoVector2F_t& gv) {
-  NovelRT::Maths::GeoVector2<float> geo = NovelRT::Maths::GeoVector2<float>(gv.x, gv.y);
-  return geo.getLength();
-}
-
-inline float GeoVector2F_getMagnitude(GeoVector2F_t& gv) {
-  return GeoVector2F_getLength(gv);
-}
-
-inline bool GeoVector2F_equal(const GeoVector2F_t& first, const GeoVector2F_t& other) {
-  return NovelRT::Maths::GeoVector2<float>(first.x, first.y) == NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-}
-
-inline bool GeoVector2F_notEqual(const GeoVector2F_t& first, const GeoVector2F_t& other) {
-  return NovelRT::Maths::GeoVector2<float>(first.x, first.y) != NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-}
-
-inline bool GeoVector2F_lessThan(const GeoVector2F_t& first, const GeoVector2F_t& other) {
-  return NovelRT::Maths::GeoVector2<float>(first.x, first.y) < NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-}
-
-inline bool GeoVector2F_lessThanOrEqualTo(const GeoVector2F_t& first, const GeoVector2F_t& other) {
-  return NovelRT::Maths::GeoVector2<float>(first.x, first.y) <= NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-}
-
-inline bool GeoVector2F_greaterThan(const GeoVector2F_t& first, const GeoVector2F_t& other) {
-  return NovelRT::Maths::GeoVector2<float>(first.x, first.y) > NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-}
-
-inline bool GeoVector2F_greaterThanOrEqualTo(const GeoVector2F_t& first, const GeoVector2F_t& other) {
-  return NovelRT::Maths::GeoVector2<float>(first.x, first.y) >= NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-}
-
-inline GeoVector2F_t GeoVector2F_addVector(GeoVector2F_t& first, const GeoVector2F_t& other) {
-  NovelRT::Maths::GeoVector2 geo = NovelRT::Maths::GeoVector2<float>(first.x,first.y) + NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-  first.x = geo.getX();
-  first.y = geo.getY();
-  return first;
-}
-
-inline GeoVector2F_t GeoVector2F_subtractVector(GeoVector2F_t& first, const GeoVector2F_t& other) {
-  NovelRT::Maths::GeoVector2 geo = NovelRT::Maths::GeoVector2<float>(first.x, first.y) - NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-  first.x = geo.getX();
-  first.y = geo.getY();
-  return first;
-}
-
-inline GeoVector2F_t GeoVector2F_multiplyVector(GeoVector2F_t& first, const GeoVector2F_t& other) {
-  NovelRT::Maths::GeoVector2 geo = NovelRT::Maths::GeoVector2<float>(first.x, first.y) * NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-  first.x = geo.getX();
-  first.y = geo.getY();
-  return first;
-}
+  using namespace NovelRT;
 
-inline GeoVector2F_t GeoVector2F_divideVector(GeoVector2F_t& first, const GeoVector2F_t& other) {
-  NovelRT::Maths::GeoVector2 geo = NovelRT::Maths::GeoVector2<float>(first.x, first.y) / NovelRT::Maths::GeoVector2<float>(other.x, other.y);
-  first.x = geo.getX();
-  first.y = geo.getY();
-  return first;
-}
+  typedef struct Maths::GeoVector2<float> GeoVector2F_t;
```

Comment:
So, I did a _bunch_ of reading and found that typically (depending on how we handle the C code), this is generally okay. With that being said, it does allow for much easier access to the C++ API in the sense that we can use the C++ objects from C, and still call the functions from the class. 

Not sure how we feel about this @tannergooding